### PR TITLE
Add helper functions to sign authorization entries.

### LINKED ISSUE #489: Add support for Auth Next.
### What problem does your feature solve?
We need to add an interface for users to add signatures in the `InvokeHostFunctionOperation.auth`.

### What would you like to see?

I have added [an implementation](https://github.com/StellarCN/py-stellar-base/blob/48b55f32fa2eb8a5418afae48a51f3234936287e/stellar_sdk/soroban/authorization_entry.py) in the Python SDK, but I'm not sure if it is the optimal solution. I would like to have more discussion on this.

Related discussion: https://github.com/stellar/js-stellar-base/pull/632
### What alternatives are there?
None

Estimated working hours: TBD, more discussion is needed.

### PR DESCRIPTION
**Pull request by overcat**
Closes #489

The following is an example.
```java
  public void sign()
      throws SorobanRpcErrorResponse,
          IOException,
          AccountNotFoundException,
          PrepareTransactionException {
    SorobanServer sorobanServer = new SorobanServer("http://127.0.0.1:8000/soroban/rpc");
    Network network = Network.STANDALONE;
    String contractId = "CDCYWK73YTYFJZZSJ5V7EDFNHYBG4QN3VUNG2IGD27KJDDPNCZKBCBXK";
    KeyPair sourceKp =
        KeyPair.fromSecretSeed("SAAPYAPTTRZMCUZFPG3G66V4ZMHTK4TWA6NS7U4F7Z3IMUD52EK4DDEV");
    KeyPair opInvokeKp =
        KeyPair.fromSecretSeed("SAEZSI6DY7AXJFIYA4PM6SIBNEYYXIEM2MSOTHFGKHDW32MBQ7KVO6EN");

    InvokeHostFunctionOperation op =
        InvokeHostFunctionOperation.invokeContractFunctionOperationBuilder(
                contractId,
                "increment",
                Arrays.asList(Scv.toAddress(opInvokeKp.getAccountId()), Scv.toUint32(10)))
            .build();

    TransactionBuilderAccount source = sorobanServer.getAccount(sourceKp.getAccountId());

    Transaction transaction =
        new TransactionBuilder(source, network)
            .addOperation(op)
            .addPreconditions(
                TransactionPreconditions.builder()
                    .timeBounds(
                        new TimeBounds(BigInteger.ZERO, TransactionPreconditions.TIMEOUT_INFINITE))
                    .build())
            .setBaseFee(50000)
            .build();

    System.out.println(transaction.toEnvelopeXdrBase64());

    SimulateTransactionResponse simulateTransactionResponse =
        sorobanServer.simulateTransaction(transaction);
    // you need to check simulateTransactionResponse
    InvokeHostFunctionOperation invokeOp =
        (InvokeHostFunctionOperation) transaction.getOperations()[0];
    // add all auth to invoke host func op
    simulateTransactionResponse
        .getResults()
        .get(0)
        .getAuth()
        .forEach(
            auth -> {
              SorobanAuthorizationEntry e =
                  Auth.authorizeEntry(
                      auth, opInvokeKp, simulateTransactionResponse.getLatestLedger() + 3, network);
              invokeOp.getAuth().add(e);
            });

    Transaction preparedTx =
        sorobanServer.prepareTransaction(transaction, simulateTransactionResponse);
    preparedTx.sign(sourceKp);
    System.out.println(preparedTx.toEnvelopeXdrBase64());

    SendTransactionResponse sendTransaction = sorobanServer.sendTransaction(preparedTx);
    System.out.println(sendTransaction);
  }
```

### CODE REVIEW

File: src/main/java/org/stellar/sdk/Auth.java [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+import org.stellar.sdk.xdr.SorobanAuthorizedInvocation;
+import org.stellar.sdk.xdr.SorobanCredentials;
+import org.stellar.sdk.xdr.SorobanCredentialsType;
+import org.stellar.sdk.xdr.Uint32;
+import org.stellar.sdk.xdr.XdrUnsignedInteger;
+
+/** This class contains helper methods to sign {@link SorobanAuthorizationEntry}. */
+public class Auth {
+  /**
+   * Actually authorizes an existing authorization entry using the given the credentials and
```
Comment (sreuland) [inline src/main/java/org/stellar/sdk/Auth.java:24]: for consideration in a later pr, could remove this duplicated docs by using `@see`  instead:
```
/**
 * @see Auth#authorizeEntry(String, KeyPair, Long, Network) authorizeEntry
 */
```

File: src/main/java/org/stellar/sdk/Auth.java [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+                            .signature(Scv.toVoid())
+                            .build())
+                    .build())
+            .rootInvocation(invocation)
+            .build();
+    return authorizeEntry(entry, signer, validUntilLedgerSeq, network);
+  }
+
+  /** An interface for signing a {@link HashIDPreimage} to produce a signature. */
+  public interface Signer {
```
Comment (sreuland) [inline src/main/java/org/stellar/sdk/Auth.java:300]: nice usage of functional interface!

File: src/test/java/org/stellar/sdk/AuthTest.java [hunk 3]
```diff
@@ -0,0 +1,606 @@
...
+        new SorobanAuthorizationEntry.Builder()
+            .credentials(expectedCredentials)
+            .rootInvocation(invocation)
+            .build();
+    assertEquals(expectedEntry, signedEntry);
+    assertNotSame(entry, signedEntry);
+  }
+
+  @Test
+  public void testSignAuthorizeEntryWithBase64EntryAndFunctionSigner() throws IOException {
```
Comment (sreuland) [inline src/test/java/org/stellar/sdk/AuthTest.java:209]: for consideration later, the duplicate test case variants for base64 and xdr object graph parameters could probably be reduced to just the base64 parameters, as that will by default exercise the xdr object path internally as well correct? would just. be less test code to maintain over time.

**Review by sreuland — APPROVED** [review]
great work, lot of complexity to auth, looks clean.

File: src/test/java/org/stellar/sdk/AuthTest.java (hunk 3 above)
Comment (overcat) [reply src/test/java/org/stellar/sdk/AuthTest.java:209]: Yes, I will remove it later.

**Pull request merged**